src/layman/layer/rest_workspace_test.py

```diff
@@ -1115,37 +1115,29 @@ def test_delete_layer(client):
 
 
 @pytest.mark.usefixtures('app_context', 'ensure_layman')
-def test_post_layers_zero_length_attribute(client):
-    username = 'testuser1'
-    rest_path = url_for('rest_workspace_layers.post', workspace=username)
+def test_post_layers_zero_length_attribute():
+    workspace = 'testuser1'
+    layername = 'zero_length_attribute'
     file_paths = [
         'sample/data/zero_length_attribute.geojson',
     ]
-    for fp in file_paths:
-        assert os.path.isfile(fp)
-    files = []
-    try:
-        files = [(open(fp, 'rb'), os.path.basename(fp)) for fp in file_paths]
-        rv = client.post(rest_path, data={
-            'file': files,
-        })
-        assert rv.status_code == 200
-    finally:
-        for fp in files:
-            fp[0].close()
 
-    layername = 'zero_length_attribute'
+    def wait_for_db_finish(response):
+        info = response.json()
+        if 'status' in info['db_table'] and info['db_table']['status'] in ['PENDING', 'STARTED']:
```

Comment:
`result = not ('status' in info['db_table'] and info['db_table']['status'] in ['PENDING', 'STARTED'])`